nautobot/extras/models/groups.py

```diff
@@ -312,13 +315,73 @@ def get_queryset(self):
 
     @property
     def members(self):
-        """Return the member objects for this group."""
+        """Return the member objects for this group, never cached."""
         # If there are child groups, return the generated group queryset, otherwise use this group's
         # `filter` directly.
         if self.children.exists():
             return self.get_group_queryset()
         return self.get_queryset()
 
+    @property
+    def members_cached(self):
+        """Return the member objects for this group, cached if available."""
+        cache_key = f"{self.__class__.__name__}.{self.id}.cached_members"
+
+        unpickled_query = None
+        try:
+            cached_query = cache.get(cache_key)
+            if cached_query is not None:
+                unpickled_query = pickle.loads(cached_query)
+        except pickle.UnpicklingError:
+            logger.warning("Failed to unpickle cached members for %s", self)
+        finally:
+            if unpickled_query is None:
+                unpickled_query = self.members.all()
+                cached_query = pickle.dumps(unpickled_query)  # Explicitly pickle the query to evaluate it.
+                cache.set(cache_key, cached_query, get_settings_or_config("DYNAMIC_GROUPS_MEMBER_CACHE_TIMEOUT"))
+
+        return unpickled_query
```

Comment:
Perhaps?

```suggestion
        try:
            cached_query = cache.get(cache_key)
            if cached_query is not None:
                return pickle.loads(cached_query)
        except pickle.UnpicklingError:
            logger.warning("Failed to unpickle cached members for %s", self)

        unpickled_query = self.members.all()
        cached_query = pickle.dumps(unpickled_query)  # Explicitly pickle the query to evaluate it.
        cache.set(cache_key, cached_query, get_settings_or_config("DYNAMIC_GROUPS_MEMBER_CACHE_TIMEOUT"))

        return unpickled_query
```

Reply:
This will cause it to always evaluate the query if I'm not mistaken, negating the cache. The former ensures we call it if unpickling was unsuccessful.

Reply:
Not sure I see that, but I guess my change would cause problems if `cached_query` unpickles to `None` (is that likely?) Fine to leave as-is.